tremor-pipeline/src/op/trickle/select.rs

```diff
@@ -603,2059 +214,261 @@ impl Operator for TrickleSelect {
-                            (group_str, group_value)
-                        })
-                        .collect();
-
-                    for (group_str, group_value) in distinct_groups {
-                        consts.group = group_value.clone_static();
-                        consts.group.push(group_str.clone())?;
-
-                        if let Some((scratch1, scratch2)) = aggregate_scratches.as_mut() {
-                            // scratches for passing on the event id between windows to track all events contributing to an outgoing windowed event
-                            let mut event_id_scratch1 = EventId::default();
-                            let mut event_id_scratch2 = EventId::default();
-
-                            // gather window events
-                            let mut emit_window_events = Vec::with_capacity(windows.len());
-                            let step1_iter = windows.iter_mut();
-                            for window in step1_iter {
-                                consts.window = Value::from(window.name.to_string());
-                                // get current window group
-                                let this_group = stry!(get_or_create_group(
-                                    window,
-                                    event_id_gen,
-                                    aggregates,
-                                    group_str,
-                                    group_value,
-                                ));
-                                let window_event = stry!(this_group.window.on_tick(ingest_ns));
-                                if window_event.emit {
-                                    emit_window_events.push(window_event);
-                                } else {
-                                    break;
-                                }
-                            }
-
-                            // merge previous aggr state, push window event out, re-initialize for new window
-                            let mut first = true;
-                            let emit_window_iter =
-                                emit_window_events.iter().zip(windows.iter_mut());
-                            for (window_event, window) in emit_window_iter {
-                                consts.window = Value::from(window.name.to_string());
+            consts.window = Value::const_null();
+            consts.group = Value::const_null();
+            consts.args = Value::const_null();
 
-                                // get current window group
-                                let this_group = stry!(get_or_create_group(
-                                    window,
-                                    event_id_gen,
-                                    aggregates,
-                                    group_str,
-                                    group_value,
-                                ));
+            let ctx = EventContext::new(ingest_ns, None);
 
```

Comment:
we should set the cartdinality here, right?

Reply:
Yes! Good catch :)